### DISCUSSION TITLE
Create new relations from the parent screen

### DISCUSSION THREAD

**Discussion by joshhornby** [orchidsoftware/platform#2610] [Q&A]
Hi,

Just want to sanity check with the community, I'm looking to offer my users the ability to create relations inline from the parent screen.

In this example I have 2 models, `Product` and `ProductVariants`

On the `Product` model I have a relationship like so

```
    public function productVariants()
    {
        return $this->hasMany(ProductVariants::class);
    }
```

On my Orchid `ProductScreen` I want to allow users to create new `ProductVariants` whilst they're viewing / editing a certain product.

**So my question is what's the correct way of doing this in Orchid?**

I could use the `Matrix` table but this doesn't quite feel right as handling when a user removes a variant and then saving this back to the DB would be a little tricky. 

The `Relation` field looks good, but doesn't offer an easy way to create a new relation.

Thanks

**Comment by c-gross** [comment]
I have used the Relations field for something like that. Saving the relation is not difficult. Saving the product and then sync the relation data.

In my case I was assign tags from a different table.

```php
$product->save();
$product->tags()->sync($tags);
```
